pygpu/gpuarray.pyx

```diff
@@ -1437,10 +1437,12 @@ cdef GpuArray pygpu_reshape(GpuArray a, unsigned int nd, const size_t *newdims,
             d = a.size // tot
 
             if d * tot != a.size:
+                free(cdims)
                 raise GpuArrayException, "..."
         cdims[i] = d
 
     array_reshape(res, a, nd, cdims, ord, nocopy)
+    free(cdims)
```

Comment:
This whole free thing should probably be in a try: .. finally since there could be an exception at basically any point that will take us out of the function.

Reply:
@abergeron I made a new commit to use try-finally block: https://github.com/Theano/libgpuarray/pull/526/commits/714f3a54e69e86ad7e71e22439bf0b06a328a2e9